packages/databricks-vscode/src/configuration/AzureCliCheck.ts

```diff
@@ -73,26 +98,43 @@ export class AzureCliCheck implements Disposable {
     public async check(silent = false): Promise<boolean> {
         this.tenantId = this.authProvider.tenantId;
 
+        let loginAttempts = 0;
+
         const steps: Record<AzureStepName, Step<boolean, AzureStepName>> = {
             tryLogin: async () => {
+                loginAttempts += 1;
                 const result = await this.tryLogin(this.authProvider.host);
-                if (typeof result === "string") {
-                    this.tenantId = result;
+                if (result.iss) {
+                    this.tenantId = result.iss;
                     return {
                         type: "next",
                         next: "loginAzureCli",
                     };
-                }
-                if (result) {
+                } else if (result.aud) {
+                    this.azureLoginAppId = result.aud;
+                    return {
+                        type: "next",
+                        next: "tryLogin",
+                    };
+                } else if (result.canLogin) {
                     return {
                         type: "success",
                         result: true,
                     };
                 } else {
-                    return {
-                        type: "next",
-                        next: "findCli",
-                    };
+                    if (loginAttempts >= 3) {
```

Comment:
We should increase this. It can reach 3 very quickly
loginAttempt = 1 : App ID incorrect
loginAttempt = 2 : tenant ID incorrect
loginAttempt = 3 : Throws error here

Reply:
bumpted it to 4